Shared/Samples/Add custom dynamic entity data source/AddCustomDynamicEntityDataSourceView.swift

```diff
@@ -0,0 +1,164 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct AddCustomDynamicEntityDataSourceView: View {
+    /// A map with an ArcGIS oceans basemap style.
+    @State var map: Map = {
+        let map = Map(basemapStyle: .arcGISOceans)
+        map.initialViewpoint = Viewpoint(
+            latitude: 47.984,
+            longitude: -123.657,
+            scale: 3e6
+        )
+        return map
+    }()
+    
+    /// The dynamic entity layer that is displaying our custom data.
+    @State var dynamicEntityLayer: DynamicEntityLayer
```

Comment:
This can be a computed property, and the state variable initialization below can be removed.
```suggestion
    private var dynamicEntityLayer: DynamicEntityLayer {
        map.operationalLayers.first as! DynamicEntityLayer
    }
```

Reply:
Based on this [comment](https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/243#discussion_r1296506907), it makes since to remove the init so I'll implement the suggestion over creating a model object.